src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -551,8 +559,8 @@ namespace AppInstaller::Logging
         }
         else
         {
-            static TelemetryTraceLogger processGlobalTelemetry;
-            return processGlobalTelemetry;
+            static GlobalTelemetryTraceLogger processGlobalTelemetry;
```

Comment:
You made things so much more complicated than necessary, and still don't seem to be calling `Initialize`.  This would have been sufficient (no header involvement; this type doesn't need to be public):
```C++
struct GlobalTelemetryTraceLogger : public TelemetryTraceLogger
{
    GlobalTelemetryTraceLogger() { Initialize(); }
}